app/providers/atmosphere/optimization_strategy/default.rb

```diff
@@ -194,12 +210,13 @@ def get_optimal_flavor_for_tenant(tmpl, t)
         end
 
         def matching_flavors(tmpl, t)
-          t.virtual_machine_flavors.active.select do |f|
-            f.supports_architecture?(tmpl.architecture) &&
-              f.memory >= min_mem &&
-              f.cpu >= min_cpu &&
-              f.hdd >= min_hdd
-          end
+          fl = Atmosphere::VirtualMachineFlavor.arel_table
+          Atmosphere::VirtualMachineFlavor.active.where(
+            fl[:tenant_id].eq(t.id).and(fl[:memory].gteq(min_mem))
+              .and(fl[:cpu].gteq(min_cpu)).and(fl[:hdd].gteq(min_hdd))
+          .and(fl[:supported_architectures]
+            .in([tmpl.architecture, 'i386_and_x86_64'])))
+          .includes(flavor_os_families: :os_family)
```

Comment:
Place the . on the previous line, together with the method call receiver.<br>Use 2 (not 0) spaces for indenting an expression spanning multiple lines.